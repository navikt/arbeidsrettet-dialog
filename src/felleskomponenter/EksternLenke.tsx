import React from 'react';
import Lenke from 'nav-frontend-lenker';
import EksternLenkeIkon from './EksternLenkeIkon';

const httpRegex = /^(https?):\/\/.*$/;

interface PropTypes {
    lenke: string | null;
}

export default function EksternLenke(props: PropTypes) {
    const { lenke } = props;

    if (!lenke) return null;
    const paddaLenke = lenke && lenke.match(httpRegex) ? lenke : `http://${lenke}`;
    return (
        <Lenke href={paddaLenke}>
            {lenke}
            <EksternLenkeIkon />
        </Lenke>
    );
}
