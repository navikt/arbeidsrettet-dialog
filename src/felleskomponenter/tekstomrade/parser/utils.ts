import { AST, ASTNode } from './domain';

export type RegexMatch = {
    fullmatch: string;
    capture: Array<string>;
    index: number;
    input: string;
};

export function match(regex: RegExp, content: string): null | RegexMatch {
    const res = regex.exec(content);
    if (!res) {
        return null;
    }
    const [fullmatch, ...capture] = res;
    return {
        fullmatch,
        capture,
        index: res.index,
        input: res.input
    };
}

export function minBy<T>(fn: (t: T) => number) {
    return (acc: null | T, value: T): null | T => {
        if (acc === null) {
            return value;
        }
        const accFn = fn(acc);
        const valueFn = fn(value);

        return accFn < valueFn ? acc : value;
    };
}

export function getText(node: AST | ASTNode): string {
    if (typeof node === 'string') {
        return node;
    } else if (Array.isArray(node)) {
        return node.map(getText).join(' ');
    } else {
        return node.content.map(getText).join(' ');
    }
}
