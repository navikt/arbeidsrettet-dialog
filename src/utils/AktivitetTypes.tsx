import { NumberOrNull, StringOrNull } from './Typer';

export enum AktivitetTypes {
    EGEN = 'EGEN',
    STILLING = 'STILLING',
    SOKEAVTALE = 'SOKEAVTALE',
    IJOBB = 'IJOBB',
    BEHANDLING = 'BEHANDLING',
    MOTE = 'MOTE',
    SAMTALEREFERAT = 'SAMTALEREFERAT'
}

export enum KanalTypes {
    OPPMOTE = 'OPPMOTE',
    TELEFON = 'TELEFON',
    INTERNETT = 'INTERNETT'
}

export interface Aktivitet {
    id: StringOrNull;
    versjon: StringOrNull;

    tittel: StringOrNull;
    beskrivelse: StringOrNull;
    lenke: StringOrNull;
    type: AktivitetTypes;
    status: StringOrNull;
    fraDato: StringOrNull;
    tilDato: StringOrNull;
    opprettetDato: string;
    endretDato: StringOrNull;
    endretAv: StringOrNull;
    historisk: boolean;
    avsluttetKommentar: StringOrNull;
    avtalt: boolean;
    lagtInnAv: StringOrNull;
    transaksjonsType: StringOrNull;
    malid: StringOrNull;

    // stillingaktivitet
    etikett: StringOrNull;
    kontaktperson: StringOrNull;
    arbeidsgiver: StringOrNull;
    arbeidssted: StringOrNull;
    stillingsTittel: StringOrNull;

    // egenaktivitet
    hensikt: StringOrNull;
    oppfolging: StringOrNull;

    //sokeAvtaleAktivitet
    antallStillingerSokes: NumberOrNull;
    antallStillingerIUken: NumberOrNull;
    avtaleOppfolging: StringOrNull;

    //iJobbAktivitet
    jobbStatus: StringOrNull;
    ansettelsesforhold: StringOrNull;
    arbeidstid: StringOrNull;

    //behandlingAktivitet
    behandlingType: StringOrNull;
    behandlingSted: StringOrNull;
    effekt: StringOrNull;
    behandlingOppfolging: StringOrNull;

    //m√∏te
    adresse: StringOrNull;
    forberedelser: StringOrNull;
    kanal: KanalTypes;
    referat: StringOrNull;
    erReferatPublisert: boolean;
}

export enum ArenaAktivitetTypes {
    TILTAKSAKTIVITET = 'TILTAKSAKTIVITET',
    GRUPPEAKTIVITET = 'GRUPPEAKTIVITET',
    UTDANNINGSAKTIVITET = 'UTDANNING'
}

export interface ArenaAktivitet {
    //Felles
    id: StringOrNull;
    status: StringOrNull;
    type: ArenaAktivitetTypes;
    tittel: StringOrNull;
    beskrivelse: StringOrNull;
    fraDato: StringOrNull;
    tilDato: StringOrNull;
    opprettetDato: StringOrNull;
    avtalt: boolean;
    etikett: StringOrNull;

    // Tiltaksaktivitet
    deltakelseProsent: NumberOrNull;
    tiltaksnavn: StringOrNull;
    tiltakLokaltNavn: StringOrNull;
    arrangoer: StringOrNull;
    bedriftsnummer: StringOrNull;
    antallDagerPerUke: NumberOrNull;
    statusSistEndret: StringOrNull;

    // Gruppeaktivitet
    moeteplanListe: object[] | null;
}
