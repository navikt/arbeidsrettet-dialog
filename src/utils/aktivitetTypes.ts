import { NumberOrNull, StringOrNull } from './Typer';

export enum AktivitetTypes {
    EGEN = 'EGEN',
    STILLING = 'STILLING',
    SOKEAVTALE = 'SOKEAVTALE',
    IJOBB = 'IJOBB',
    BEHANDLING = 'BEHANDLING',
    MOTE = 'MOTE',
    SAMTALEREFERAT = 'SAMTALEREFERAT',
    STILLING_FRA_NAV = 'STILLING_FRA_NAV',
    EKSTERN_AKTIVITET = 'EKSTERNAKTIVITET'
}

export type AlleAktivitetTypes =
    | Exclude<AktivitetTypes, AktivitetTypes.EKSTERN_AKTIVITET>
    | ArenaAktivitetTypes
    | EksternAktivitetTypes;

export enum KanalTypes {
    OPPMOTE = 'OPPMOTE',
    TELEFON = 'TELEFON',
    INTERNETT = 'INTERNETT'
}

export enum AktivitetStatus {
    PLANLAGT = 'PLANLAGT',
    GJENNOMFORES = 'GJENNOMFORES',
    FULLFORT = 'FULLFORT',
    BRUKER_ER_INTERESSERT = 'BRUKER_ER_INTERESSERT',
    AVBRUTT = 'AVBRUTT'
}

export type StillingStatus =
    | 'INGEN_VALGT'
    | 'SOKNAD_SENDT'
    | 'INNKALT_TIL_INTERVJU'
    | 'AVSLAG'
    | 'JOBBTILBUD'
    | null
    | undefined;

export interface CvKanDelesData {
    kanDeles: boolean;
    endretTidspunkt: StringOrNull;
    avtaltDato: StringOrNull;
    endretAv: StringOrNull;
    endretAvType: BrukerType;
}
export type BrukerType = 'NAV' | 'BRUKER';

export enum StillingFraNavSoknadsstatus {
    VENTER = 'VENTER',
    SKAL_PAA_INTERVJU = 'SKAL_PAA_INTERVJU',
    JOBBTILBUD = 'JOBBTILBUD',
    AVSLAG = 'AVSLAG',
    CV_DELT = 'CV_DELT',
    IKKE_FATT_JOBBEN = 'IKKE_FATT_JOBBEN',
    FATT_JOBBEN = 'FATT_JOBBEN'
}
export interface StillingFraNavData {
    cvKanDelesData: CvKanDelesData;
    soknadsfrist: StringOrNull;
    svarfrist: StringOrNull;
    arbeidsgiver: StringOrNull;
    bestillingsId: string;
    stillingsId: StringOrNull;
    arbeidssted: StringOrNull;
    varselId: StringOrNull;
    lenke: StringOrNull; //mangler i backend
    soknadsstatus: StillingFraNavSoknadsstatus;
    //    livslopsstatus: Livslopsstatus;
}

export interface Aktivitet {
    id: string;
    versjon: StringOrNull;

    tittel: StringOrNull;
    beskrivelse: StringOrNull;
    lenke: StringOrNull;
    type: AktivitetTypes;
    status: AktivitetStatus;
    fraDato: StringOrNull;
    tilDato: StringOrNull;
    opprettetDato: string;
    endretDato: StringOrNull;
    endretAv: StringOrNull;
    historisk: boolean;
    avsluttetKommentar: StringOrNull;
    avtalt: boolean;
    endretAvType: StringOrNull;
    transaksjonsType: StringOrNull;
    malid: StringOrNull;
    oppfolgingsperiodeId: string;

    // stillingaktivitet
    etikett: StillingStatus;
    kontaktperson: StringOrNull;
    arbeidsgiver: StringOrNull;
    arbeidssted: StringOrNull;
    stillingsTittel: StringOrNull;

    // egenaktivitet
    hensikt: StringOrNull;
    oppfolging: StringOrNull;

    //sokeAvtaleAktivitet
    antallStillingerSokes: NumberOrNull;
    antallStillingerIUken: NumberOrNull;
    avtaleOppfolging: StringOrNull;

    //iJobbAktivitet
    jobbStatus: StringOrNull;
    ansettelsesforhold: StringOrNull;
    arbeidstid: StringOrNull;

    //behandlingAktivitet
    behandlingType: StringOrNull;
    behandlingSted: StringOrNull;
    effekt: StringOrNull;
    behandlingOppfolging: StringOrNull;

    //m√∏te
    adresse: StringOrNull;
    forberedelser: StringOrNull;
    kanal?: KanalTypes;
    referat: StringOrNull;
    erReferatPublisert: boolean;

    // stillingFraNav
    stillingFraNavData?: StillingFraNavData;

    // eksternaktivitet
    eksternAktivitet?: EksternAktivitetData;
}

export interface EksternAktivitetData {
    type: EksternAktivitetTypes;
}

export enum EksternAktivitetTypes {
    ARENA_TILTAK = 'ARENA_TILTAK',
    MIDLERTIDIG_LONNSTILSKUDD = 'MIDLERTIDIG_LONNSTILSKUDD',
    VARIG_LONNSTILSKUDD = 'VARIG_LONNSTILSKUDD',
    INDOPPFAG = 'INDOPPFAG',
    ARBFORB = 'ARBFORB',
    AVKLARAG = 'AVKLARAG',
    VASV = 'VASV',
    ARBRRHDAG = 'ARBRRHDAG',
    DIGIOPPARB = 'DIGIOPPARB',
    JOBBK = 'JOBBK',
    GRUPPEAMO = 'GRUPPEAMO',
    GRUFAGYRKE = 'GRUFAGYRKE'
}

export enum ArenaAktivitetTypes {
    TILTAKSAKTIVITET = 'TILTAKSAKTIVITET',
    GRUPPEAKTIVITET = 'GRUPPEAKTIVITET',
    UTDANNINGSAKTIVITET = 'UTDANNINGSAKTIVITET'
}

export interface ArenaAktivitet {
    //Felles
    id: string;
    status: AktivitetStatus;
    type: ArenaAktivitetTypes;
    tittel: StringOrNull;
    beskrivelse: StringOrNull;
    fraDato: StringOrNull;
    tilDato: StringOrNull;
    opprettetDato: StringOrNull;
    avtalt: boolean;
    etikett: StringOrNull;

    // Tiltaksaktivitet
    deltakelseProsent: NumberOrNull;
    tiltaksnavn: StringOrNull;
    tiltakLokaltNavn: StringOrNull;
    arrangoer: StringOrNull;
    bedriftsnummer: StringOrNull;
    antallDagerPerUke: NumberOrNull;
    statusSistEndret: StringOrNull;

    // Gruppeaktivitet
    moeteplanListe: object[] | null;
}
