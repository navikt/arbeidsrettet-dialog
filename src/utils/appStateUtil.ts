import { AppState } from '../view/Provider';
import { Status } from '../api/typer';

const key = 'dialog-state';
export const getPreloadedStateFromSessionStorage = (fnr: string | undefined): AppState | undefined => {
    if (!fnr) return undefined;
    const serializedState = sessionStorage.getItem(key);
    if (serializedState) {
        try {
            const state: AppState = JSON.parse(serializedState);
            // Only use cache if correct user
            if (fnr === state.data.oppfolging?.data?.fnr) {
                return JSON.parse(serializedState);
            }
            sessionStorage.removeItem(key);
            return undefined;
        } catch (e) {
            console.warn(e);
            return undefined;
        }
    }
    return undefined;
};

export function localStorageProvider() {
    // When initializing, we restore the data from `localStorage` into a map.
    const map = new Map(JSON.parse(localStorage.getItem('app-cache') || '[]'));

    // Before unloading the app, we write back all the data into `localStorage`.
    window.addEventListener('beforeunload', () => {
        const appCache = JSON.stringify(Array.from(map.entries()));
        localStorage.setItem('app-cache', appCache);
    });

    // We still use the map for write & read for performance.
    return map;
}

enum LocalStorageElement {
    FNR = 'dialog-fnr'
}
export const settFnrILocalstorage = (fnr: string) => {};

export const getStatus = (data: any, isLoading: boolean, isValidating: boolean, error: any) => {
    if (isLoading) return Status.PENDING;
    if (isValidating) return Status.RELOADING;
    if (error) return Status.ERROR;
    if (!data) return Status.INITIAL;
    return Status.OK;
};
