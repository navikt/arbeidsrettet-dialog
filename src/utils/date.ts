import { format } from 'date-fns';
import * as norweigianLocale from 'date-fns/locale/nb';

export function formaterDateAndTime(sendtDate: string) {
    return format(new Date(sendtDate), 'DD.MM.YYYY HH:mm', { locale: norweigianLocale });
}

export function formaterDate(date: string) {
    return formaterDateAndTime(date).substring(0, 10);
}
