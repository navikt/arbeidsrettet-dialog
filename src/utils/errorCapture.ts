/* Capture error without importing sentry */

/* Make sure to not import sentry for types, just create a similar type */
declare const window: {
    captureException: (exception: any) => void;
    captureMessage: (message: string) => void;
};

export const captureException = (exception: Error) => {
    if (window.captureException) {
        window.captureException(exception);
    }
};
export const captureMessage = (message: string) => {
    if (window.captureMessage) {
        window.captureMessage(message);
    }
};

export const captureMaybeError = (errorMessage: string, error: Error | any) => {
    if (error instanceof Error) {
        captureException(new Error(errorMessage, { cause: error }));
    } else {
        captureMessage(errorMessage);
    }
};
