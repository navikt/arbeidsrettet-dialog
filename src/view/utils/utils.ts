import { Aktivitet, AktivitetTypes, AlleAktivitetTypes, ArenaAktivitet } from '../../utils/aktivitetTypes';
import { pathnamePrefix } from '../../utils/UseApiBasePath';

export const erArenaAktivitet = (aktivitetId: string | null | undefined): boolean =>
    !!aktivitetId && aktivitetId.startsWith('ARENA');

export const getBasename = (fnr?: string): string => {
    return settSammenmedSlasher(pathnamePrefix, fnr);
};

export const settSammenmedSlasher = (...ss: Array<string | undefined>): string => {
    const slasherImellom = ss
        .filter((s: string | undefined): s is string => !!s)
        .flatMap<string>((s) => s.split('/'))
        .filter((s) => s.trim() !== '')
        .join('/');
    if (slasherImellom !== '') return '/' + slasherImellom;
    else return '';
};

const erGCP = (): boolean => window.location.hostname.endsWith('intern.nav.no');

export const getContextPath = (): string => (erGCP() ? '' : '/veilarbpersonflatefs');

export const getAktivitetType = (aktivitet: Aktivitet | ArenaAktivitet): AlleAktivitetTypes => {
    if (aktivitet.type === AktivitetTypes.EKSTERN_AKTIVITET) {
        return aktivitet.eksternAktivitet!!.type;
    }
    return aktivitet.type;
};

export const runningOnGithubPages = process.env.REACT_APP_USE_HASH_ROUTER === 'true';
