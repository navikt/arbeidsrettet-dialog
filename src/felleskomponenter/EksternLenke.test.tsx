import React from 'react';
import { shallow } from 'enzyme';
import EksternLenke from './EksternLenke';
import '../utils/SetupEnzyme';

describe('<EksternLenke/>', () => {
    it('skal padde href med protokoll når det mangler', () => {
        const tekst = 'nav.no';
        const wrapper = shallow(<EksternLenke lenke={tekst} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('skal ikke padde href med protokoll, http', () => {
        const tekst = 'http://nav.no';
        const wrapper = shallow(<EksternLenke lenke={tekst} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('skal ikke padde href med protokoll, https', () => {
        const tekst = 'https://nav.no';
        const wrapper = shallow(<EksternLenke lenke={tekst} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('skal padde href med protokoll når det mangler, www', () => {
        const tekst = 'www.nav.no';
        const wrapper = shallow(<EksternLenke lenke={tekst} />);

        expect(wrapper).toMatchSnapshot();
    });
});
