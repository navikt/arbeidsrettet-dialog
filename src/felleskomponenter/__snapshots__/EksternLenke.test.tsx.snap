// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<EksternLenke/> > skal ikke padde href med protokoll, http 1`] = `
<body>
  <div>
    <a
      class="navds-link navds-link--action"
      data-color="accent"
      data-variant="action"
      href="http://nav.no"
    >
      http://nav.no
      <svg
        height="14px"
        viewBox="0 0 14 14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          id="ikon_external_link"
        >
          <g
            id="LINE/interface/-external-link"
            stroke="#0067C5"
            transform="translate(2 2)"
          >
            <g
              id="external-link"
            >
              <path
                d="M10,7.07712912 L10,8.33333333 C10,9.25380792 9.25380792,10 8.33333333,10 L1.66666667,10 C0.746192084,10 0,9.25380792 0,8.33333333 L0,1.66666667 C0,0.746192084 0.746192084,0 1.66666667,0 L2.94199316,0"
                id="Rectangle-6"
              />
              <g
                id="Group-7"
                transform="translate(4)"
              >
                <path
                  d="M0,6 L6,0"
                  id="Line-6"
                />
                <polyline
                  id="Path-2"
                  points="2.00719286 0 6 0 6 4.06130668"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </a>
  </div>
</body>
`;

exports[`<EksternLenke/> > skal ikke padde href med protokoll, https 1`] = `
<body>
  <div>
    <a
      class="navds-link navds-link--action"
      data-color="accent"
      data-variant="action"
      href="https://nav.no"
    >
      https://nav.no
      <svg
        height="14px"
        viewBox="0 0 14 14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          id="ikon_external_link"
        >
          <g
            id="LINE/interface/-external-link"
            stroke="#0067C5"
            transform="translate(2 2)"
          >
            <g
              id="external-link"
            >
              <path
                d="M10,7.07712912 L10,8.33333333 C10,9.25380792 9.25380792,10 8.33333333,10 L1.66666667,10 C0.746192084,10 0,9.25380792 0,8.33333333 L0,1.66666667 C0,0.746192084 0.746192084,0 1.66666667,0 L2.94199316,0"
                id="Rectangle-6"
              />
              <g
                id="Group-7"
                transform="translate(4)"
              >
                <path
                  d="M0,6 L6,0"
                  id="Line-6"
                />
                <polyline
                  id="Path-2"
                  points="2.00719286 0 6 0 6 4.06130668"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </a>
  </div>
</body>
`;

exports[`<EksternLenke/> > skal padde href med protokoll når det mangler 1`] = `
<body>
  <div>
    <a
      class="navds-link navds-link--action"
      data-color="accent"
      data-variant="action"
      href="http://nav.no"
    >
      nav.no
      <svg
        height="14px"
        viewBox="0 0 14 14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          id="ikon_external_link"
        >
          <g
            id="LINE/interface/-external-link"
            stroke="#0067C5"
            transform="translate(2 2)"
          >
            <g
              id="external-link"
            >
              <path
                d="M10,7.07712912 L10,8.33333333 C10,9.25380792 9.25380792,10 8.33333333,10 L1.66666667,10 C0.746192084,10 0,9.25380792 0,8.33333333 L0,1.66666667 C0,0.746192084 0.746192084,0 1.66666667,0 L2.94199316,0"
                id="Rectangle-6"
              />
              <g
                id="Group-7"
                transform="translate(4)"
              >
                <path
                  d="M0,6 L6,0"
                  id="Line-6"
                />
                <polyline
                  id="Path-2"
                  points="2.00719286 0 6 0 6 4.06130668"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </a>
  </div>
</body>
`;

exports[`<EksternLenke/> > skal padde href med protokoll når det mangler, www 1`] = `
<body>
  <div>
    <a
      class="navds-link navds-link--action"
      data-color="accent"
      data-variant="action"
      href="http://www.nav.no"
    >
      www.nav.no
      <svg
        height="14px"
        viewBox="0 0 14 14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          id="ikon_external_link"
        >
          <g
            id="LINE/interface/-external-link"
            stroke="#0067C5"
            transform="translate(2 2)"
          >
            <g
              id="external-link"
            >
              <path
                d="M10,7.07712912 L10,8.33333333 C10,9.25380792 9.25380792,10 8.33333333,10 L1.66666667,10 C0.746192084,10 0,9.25380792 0,8.33333333 L0,1.66666667 C0,0.746192084 0.746192084,0 1.66666667,0 L2.94199316,0"
                id="Rectangle-6"
              />
              <g
                id="Group-7"
                transform="translate(4)"
              >
                <path
                  d="M0,6 L6,0"
                  id="Line-6"
                />
                <polyline
                  id="Path-2"
                  points="2.00719286 0 6 0 6 4.06130668"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </a>
  </div>
</body>
`;
